Cargo.toml

```diff
@@ -96,6 +96,9 @@ structopt = "0.3"
 tempfile = "3.1"
 test-case = "1.0"
 tracing-subscriber = "=0.2.12"
+async-std = "1.8.0"
+serde_json = "1"
+tokio = { version = "0.3", features = ["full"] }
```

Comment:
This is a lot to pull in just for one doc example. Why both async-std and tokio at the same time?

Reply:
Yeah,I just want use `tokio::io::copy`:
```rust
tokio::io::copy(response.body_mut(), &mut body).await.unwrap();
```

but build error, the error detail:
```bash
error[E0277]: the trait bound `Body: AsyncRead` is not satisfied
  --> src/main.rs:14:5
   |
14 |     tokio::io::copy(response.body_mut(), &mut body).await.unwrap();
   |     ^^^^^^^^^^^^^^^ the trait `AsyncRead` is not implemented for `Body`
   | 
  ::: /Users/baoyachi/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.3.5/src/io/util/copy.rs:58:12
   |
58 |         R: AsyncRead + Unpin + ?Sized,
   |            --------- required by this bound in `tokio::io::copy`

error: aborting due to previous error
```
The tokio copy method need `Body` is `AsyncRead`.

But I just find isahc `Response<T>` not generic constraints. 

The `async_std::io::copy` not constraints. It's build success.

Should `Response<T>` add any generic constraints?

Reply:
I think Rust gave you a poor error message perhaps. Tokio's been doing some funky things with I/O lately, and their `AsyncRead` trait is a different one than the defacto one from the `futures` crate. They're both called `AsyncRead`, but the interface is different, which is why it is confusing.

You could use `futures_lite::io::copy` instead, since `futures_lite` is already a dependency and works with the traits from `futures`. Though maybe that would make the example less familiar?

You can also see an example of how to avoid using something like `#[tokio::main]` in doc examples here:

https://github.com/sagebind/isahc/blob/1fdcae7f788df30ae0875926121e7bdf50b923e6/src/client.rs#L955-L970

Admittedly not pretty, but gets the job done and it looks good in rustdoc.

Reply:
Ok, Your opinion is meaningful.I just edit

Reply:
It solve by commit:https://github.com/sagebind/isahc/pull/273/commits/f6ccafaea52d6a6d662209efa1dd7f6f105668f4.